### ISSUE TITLE
Function generator: error in `yield` statement

### ISSUE THREAD

**Issue by zdychacek** [palantir/tslint#706]
Hi,

I am getting an error in `yield` statement 'Error: expected an assignment or function call' message in following code:

```
function *g (): Iterable<number> {
	for (let i: number = 0; i < 100; i++) {
		yield i;
	}
}
```

Is there really any error in my code?

Thanks for your reply,

O.

**Comment by adidahiya** [comment]
This looks like an error reported by the compiler rather than TSLint... can you post your full stack trace or console output?

**Comment by JKillian** [comment]
A problem on the TSLint side might be that we target ES5 instead of ES6. I think [generators only work in ES6.](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#es6-generators)

**Comment by zdychacek** [comment]
When I run `tslint main.ts` I get following `main.ts[3, 3]: expected an assignment or function call`, so I think this issue has something to do with TSLint.

I am targetting ES6.

Here is my `tsconfig.json`:

```
{
  "compilerOptions": {
    "module": "commonjs",
    "noImplicitAny": true,
    "target": "es6",
    "experimentalDecorators": true
  }
}
```

Hope this information helps somehow.

Thanks,

O.

**Comment by JKillian** [comment]
@zdychacek Sorry, I didn't mean that you need to be targeting ES6; I meant that TSLint ought to have an option that makes it target ES6 when analyzing your code. I don't know if that's the issue for sure though. Definitely a bug either way, thanks for reporting it!

**Comment by JKillian** [comment]
Seems like we can handle this better even when internally targeting ES5, see the above PR :)

**Comment by zdychacek** [comment]
Great, thank you!

O.

**Comment by adidahiya** [comment]
the fix will be released in v2.5.2. until then, I think you can avoid the issue with `/* tslint:disable */` flags?
